src/PackageExtensionCompat.jl

```diff
@@ -7,37 +7,63 @@ const HAS_NATIVE_EXTENSIONS = isdefined(Base, :get_extension)
 @static if !HAS_NATIVE_EXTENSIONS
     using Requires, TOML
 
-    function rewrite_import(str, pkgs)
-        parts = split(strip(str))
-        if length(parts) == 1 || (length(parts) ≥ 2 && parts[2] == "as")
-            if parts[1] ∈ pkgs
-                parts[1] = string("..", parts[1])
+    @static if hasmethod(Base.include, Tuple{Function, Module, String})
+        function _include(mapexpr::Function, m::Module, path::AbstractString)
+            Base.include(mapexpr, m, path)
+        end
+    else
+        function _include(mapexpr::Function, m::Module, path::AbstractString)
+            path = abspath(path)
+            cd(dirname(path)) do
+                str = read(basename(path), String)
+                pos = 1
+                while true
+                    (expr, pos) = Meta.parse(str, pos; raise=false)
+                    expr !== nothing || break
+                    m.eval(mapexpr(expr))
+                end
             end
         end
-        join(parts, " ")
     end
 
-    function rewrite_imports(str, pkgs)
-        parts = split(str, ",")
-        parts = map(part -> rewrite_import(part, pkgs), parts)
-        join(parts, ", ")
+    # a simplified variant of MacroTools.postwalk
+    postwalk(f, x) = f(x)
+    postwalk(f, x::Expr) = f(Expr(x.head, postwalk.(f, x.args)...))
+
+    function rewrite(top_pkg::Module, pkgs)
+        Base.Fix1(postwalk, block -> rewrite_block(block, top_pkg, pkgs))
     end
 
-    function rewrite_line(line, pkgs)
-        pat = r"^(\s*(using|import)\s+)([^;:#$]*[^;:#$\s])(.*)$"
-        m = match(pat, line)
-        if m === nothing
-            line
+    function rewrite_block(block, top_pkg::Module, pkgs)
+        if Meta.isexpr(block, :call) && block.args[1] == :include
+            # inner include, rewrite it recursively
```

Comment:
I take back an earlier comment - you do rewrite includes recursively. You should check it's being called with one argument.

Reply:
Up to julia 1.4, there is only one method in the global `include` function, and that method takes a single string as an argument. So I doubt there will be any other kind of call.

From julia 1.5, there is also a two argument variant, but the same version introduces the three arugment method of `Base.include`, and we will use that.

But if you still want this argcount check, I can add it for you.

Reply:
Yeah please check.

Reply:
Sorry, I made this change 5 days ago, just forgot to push it. Duh.